src/components/home/HomeSideBar.js

```diff
@@ -1,38 +1,22 @@
 import React from 'react';
-import { FormattedMessage } from 'react-intl';
 import { connect } from 'react-redux';
-import { string } from 'prop-types';
+import { string, object } from 'prop-types';
 
 import SmilesLogo from 'resources/icons/SmilesLogo';
 import UserMenu from 'components/user/UserMenu';
+import FirstTarget from './FirstTarget';
+import Matches from './Matches';
 
-const HomeSideBar = ({ username }) => (
+const HomeSideBar = ({ username, targets }) => (
   <div className="page-container page-container--full-height page-container--shrink show-for-medium" >
     <div className="container--spaced">
+      <div>
+        <h1 className="title mb1">Target</h1>
+        <UserMenu username={username} />
+      </div>
       <div className="container--spaced">
-        <div>
-          <h1 className="title mb1">Target</h1>
-          <UserMenu username={username} />
-        </div>
-        <h2 className="subtitle center">
-          <FormattedMessage id="home.sidebar.firstTarget" />
-        </h2>
-        <div>
-          <span className="copy copy--bold">
-            <FormattedMessage id="home.sidebar.popular" />
-          </span>
-          <div className="list">
-            <div className="list-item list-item--bullet">
-              <FormattedMessage id="home.sidebar.target1" />
-            </div>
-            <div className="list-item list-item--bullet">
-              <FormattedMessage id="home.sidebar.target2" />
-            </div>
-            <div className="list-item list-item--bullet">
-              <FormattedMessage id="home.sidebar.target3" />
-            </div>
-          </div>
-        </div>
+        {targets.size > 0 && <Matches />}
```

Comment:
You can use a ternary operator here. Also since `size` is supposed to return an integer you can just ask for `.size` being true as any value other than 0 will be true, no need to compare against zero.

```suggestion
        {targets.size ? <Matches /> : <FirstTarget />}
```